### PR TITLE
chore: update help info

### PR DESCRIPTION
**Pull request by daixiang0**
Update `make help`:
- add missing targets
- add run in docker comment as we still support it

Signed-off-by: Loong Dai <loong.dai@intel.com>

### CODE REVIEW

**Comment by daixiang0** [comment]
@arkodg please review.

File: tools/make/golang.mk [hunk 1]
```diff
@@ -65,8 +65,8 @@ go.format:  go.format.verify
 build: ## Build envoy-gateway for host platform. See Option PLATFORM and BINS.
 	@$(MAKE) go.build
 
-.PHONY: build.multiarch
-build.multiarch: ## Build envoy-gateway for multiple platforms. See Option PLATFORMS and IMAGES.
+.PHONY: build-multiarch
+build-multiarch: ## Build envoy-gateway for multiple platforms. See Option PLATFORMS and IMAGES.
```
Comment (Xunzhuo) [inline tools/make/golang.mk:69]: Any good reasons to change the `.` to `-`?

**Review by arkodg — APPROVED** [review]
LGTM thanks

File: tools/make/golang.mk (hunk 1 above)
Comment (arkodg) [reply tools/make/golang.mk:69]: it doesn't show up in `make help`, worth debugging why :)

Comment (Xunzhuo) [reply tools/make/golang.mk:69]: Got it, thanks @arkodg and @daixiang0

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**